source/data-modeling/persistence-configuration.txt

```diff
@@ -0,0 +1,252 @@
+
+By default, the ``Band`` class will store documents in a collection called
+``bands``. By using the ``with()`` method, you can temporarily perform operations 
+on the ``Band`` model class's documents using a different client, database, or 
+collection.
+
+The following example shows how to perform operations on the ``Band`` model class
+within a different client, database, and collection.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start with() example
+  :end-before: end with() example
+
+The persistence context for the model class or instance that ``with`` is called
+on is changed to the temporary persistence context defined by the arguments
+passed to the ``with()`` method within the block. The model class or instance 
+that ``with`` was called on is yielded to the block.
+
+The temporary persistence context applies to both queries and writes.
+
+Ensuring Consistency in Contexts
+++++++++++++++++++++++++++++++++
+
+If a document is saved in a temporary persistence context, it may not exist
+in the default persistence context. For example, if you insert a ``Band`` model
+document into the ``artists`` collection, the same document will not be inserted
+into the ``bands`` collection. The following example shows how to correctly 
+ensure consistency between contexts.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start consistency example
+  :end-before: end consistency example
+
+.. important:: Using Block Scope
+   
+   As of {+odm+} 6.0, the ``with`` method must always be called with a block.
+   This is because a new client is created in the background with the
+   corresponding options that are passed to ``with``. The scope when this new
+   client is used must be well-defined so the client can be closed and
+   its resources can be freed.
+
+``with`` can also apply directly to read and write operations. The following 
+example reads and counts data from the secondary node in an application:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start read configuration
+  :end-before: end read configuration
+
+Setting Persistence Contexts Globally
++++++++++++++++++++++++++++++++++++++
+
+{+odm+} provides the ability to globally define the persistence configuration at
+the database and client level. This is helpful if you want to switch the 
+persistence context for all operations at runtime but don't want to use ``with`` 
+multiple times throughout your code.
+
+To globally define the persistence context in your program, use the
+``{+odm+}.override_client`` and ``{+odm+}.override_database`` methods. For 
+example, if your application stores information for different locales in 
+different databases, but the overall schema remains the same, you
+can use the ``{+odm+}.override_database`` function:
```

Comment:
S: combine example with code intro. use 'method' instead of 'function' for consistency

```suggestion
``{+odm+}.override_client`` and ``{+odm+}.override_database`` methods.

 In the following code example, the application stores information for different locales
 in different databases, but both databases[?] use the same schema. The code shows
 how to use the ``{+odm+}.override_database()`` method to set a global persistence
 context for these databases:
```